tests/phpunit/bootstrap.php

```diff
@@ -1,28 +1,52 @@
+}
+
+if ( ! file_exists( $_tests_dir . '/includes/functions.php' ) ) {
+	echo "Could not find $_tests_dir/includes/functions.php, have you run bin/install-wp-tests.sh ?" . PHP_EOL; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+	exit( 1 );
 }
 
+if ( PHP_VERSION_ID >= 80000 && file_exists( $_tests_dir . '/includes/phpunit7/MockObject' ) ) {
+	// WP Core test library includes patches for PHPUnit 7 to make it compatible with PHP8.
+	require_once $_tests_dir . '/includes/phpunit7/MockObject/Builder/NamespaceMatch.php';
+	require_once $_tests_dir . '/includes/phpunit7/MockObject/Builder/ParametersMatch.php';
+	require_once $_tests_dir . '/includes/phpunit7/MockObject/InvocationMocker.php';
+	require_once $_tests_dir . '/includes/phpunit7/MockObject/MockMethod.php';
+}
+
+// Give access to tests_add_filter() function.
 require_once $_tests_dir . '/includes/functions.php';
 
+/**
+ * Manually load the plugin being tested.
+ */
 function _manually_load_plugin() {
-	$plugin_dir = dirname( dirname( dirname( dirname( __FILE__ ) ) ) ) . '/';
-	require $plugin_dir . 'woocommerce/woocommerce.php';
-	require dirname( __FILE__ ) . '/setup.php';
-	require $plugin_dir . 'woocommerce-gateway-stripe/woocommerce-gateway-stripe.php';
+	// Load the WooCommerce plugin so we can use its classes in our WooCommerce Payments plugin.
+	require_once ABSPATH . '/wp-content/plugins/woocommerce/woocommerce.php';
+	require __DIR__ . '/setup.php';
+
+	$_plugin_dir = __DIR__ . '/../../';
+	require $_plugin_dir . 'woocommerce-gateway-stripe.php';
 }
 
 tests_add_filter( 'muplugins_loaded', '_manually_load_plugin' );
 
 require $_tests_dir . '/includes/bootstrap.php';
 
-$wc_tests_framework_base_dir = dirname( dirname( dirname( dirname( __FILE__ ) ) ) ) . '/woocommerce/tests/framework/';
-require_once( $wc_tests_framework_base_dir . 'class-wc-mock-session-handler.php' );
-//require_once( $wc_tests_framework_base_dir . 'class-wc-unit-test-case.php' );
-require_once( $wc_tests_framework_base_dir . 'helpers/class-wc-helper-product.php' );
-require_once( $wc_tests_framework_base_dir . 'helpers/class-wc-helper-coupon.php' );
-require_once( $wc_tests_framework_base_dir . 'helpers/class-wc-helper-fee.php' );
-require_once( $wc_tests_framework_base_dir . 'helpers/class-wc-helper-shipping.php' );
-require_once( $wc_tests_framework_base_dir . 'helpers/class-wc-helper-customer.php' );
-require_once( $wc_tests_framework_base_dir . 'helpers/class-wc-helper-order.php' );
+require_once( __DIR__ . '/helpers/class-wc-mock-session-handler.php' );
+require_once( __DIR__ . '/helpers/class-wc-helper-product.php' );
+require_once( __DIR__ . '/helpers/class-wc-helper-coupon.php' );
+require_once( __DIR__ . '/helpers/class-wc-helper-fee.php' );
+require_once( __DIR__ . '/helpers/class-wc-helper-shipping.php' );
+require_once( __DIR__ . '/helpers/class-wc-helper-customer.php' );
+require_once( __DIR__ . '/helpers/class-wc-helper-order.php' );
```

Comment:
These helpers don't seem to be used anywhere:

```php
require_once( __DIR__ . '/helpers/class-wc-mock-session-handler.php' );
require_once( __DIR__ . '/helpers/class-wc-helper-coupon.php' );
require_once( __DIR__ . '/helpers/class-wc-helper-fee.php' );
require_once( __DIR__ . '/helpers/class-wc-helper-customer.php' );
```

Shall we remove them and add if necessary when writing new tests?

Reply:
Good catch! I agree we should remove them, and add them as necessary

Reply:
Perhaps we could add a comment saying where these helpers come from (the folder URL), so it's easier to import them in the future and it's clear for someone not familiar with the code.

Reply:
Done